ArduCopter/precision_landing.cpp

```diff
@@ -13,13 +13,11 @@ void Copter::init_precland()
 
 void Copter::update_precland()
 {
-    int32_t height_above_ground_cm = current_loc.alt;
-
-    // use range finder altitude if it is valid, otherwise use home alt
-    if (rangefinder_alt_ok()) {
-        height_above_ground_cm = rangefinder_state.alt_cm_glitch_protected;
+    if (!rangefinder_alt_ok() ||
+        rangefinder_state.alt_cm_glitch_protected == 0) {
```

Comment:
The, "rangefinder_state.alt_cm_glitch_protected == 0" check is only checking if there has ever been an un-glitched rangefinder altitude so I imagine it is only useful for the first few seconds after startup.  What do you think?


 right.  So the rangefinder could be glitching right now but we would still pass through the glitch protected value.

Reply:
By the way, the removal of "height_above_ground_cm = current_loc.alt" should be fine.  I noticed a long time ago that AC_PrecLand would never actually use the current_loc.alt but simply hadn't gotten around to removing it.

Reply:
Yes, should be non-zero unless the thing is really misbehaving - but we should check :-)

Reply:
maybe just remove the "rangefinder_state.alt_cm_glitch_protected == 0" check?  I don't think it's necessary..

Reply:
Removed the `== 0` check.